### ISSUE TITLE
Decide where to store config and chain data

### ISSUE THREAD

**Issue by gsalgado** [ethereum/py-evm#198]
Currently py-evm doesn't have a standard place to store its configs and chain data. We need to agree on one and start storing stuff there to avoid having to hard-code or [regenerate values on every run](https://github.com/ethereum/py-evm/pull/189#discussion_r154775751)

I believe we should use $XDG_DATA_HOME and $XDG_CONFIG_HOME as per https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

**Comment by pipermerriam** [comment]
@gsalgado I took a quick gander at that spec.  Would you mind expanding this issue to contain the following.

* What data do we need to store that is *global*
* What data do we need to store for each *chain*
* What folder structure we might use for this data.
* What base directories we would use according the the XDG spec from above.

**Comment by gsalgado** [comment]
I cannot think of any global data at this point. Can you think of anything?

Chain specific:
- block/state (one for lightchain and one for full chain) 
- private keys for accounts
- P2P node's private key (used to generate the node's unique ID and for authentication)
- peer statistics

Since all of that is actually _data_ (and not user-maintained _config_) generated/updated by py-evm, I believe it should go under _$XDG_DATA_HOME_ (which defaults to _$HOME/.local/share_).

As for the directory structure, I think the following would be a good start:

```
├── py-evm
│   └── mainnet
│       ├── chain         # LevelDB
│       ├── keys          # KeyStore, whatever format we chose
│       ├── lightchain    # LevelDB
│       ├── nodekey       # a text file containing the node's P2P private key
│       └── peers         # LevelDB
```

That also leaves room for any global data to be stored directly under _$XDG_DATA_HOME/py-evm_

**Comment by pipermerriam** [comment]
I like it and happy to learn about `$XDG_DATA_HOME` which is a standard I wasn't aware of.

**Comment by pipermerriam** [comment]
Oh, I think we might benefit from **not** implementing a keystore.  I don't want to manage user's private keys (at least not using py-evm).  I'm imagining this as something that moves into the web3.py layer with the `web3.account` API.  We can still read from keystore files and whatnot, I just don't want to do it at the py-evm level.

**Comment by pipermerriam** [comment]
This is fixed in #300 which is about to get merged.  Closing.
